services/registration/src/routes/index.ts

```diff
@@ -1,9 +1,56 @@
+import { asyncHandler } from "@api/common";
 import express from "express";
 
+import { ApplicationModel } from "src/models/application";
+import { BranchModel, BranchType } from "src/models/branch";
 import { applicationRouter } from "./application";
 import { branchRouter } from "./branch";
 
 export const defaultRouter = express.Router();
 
 defaultRouter.use("/branches", branchRouter);
 defaultRouter.use("/applications", applicationRouter);
+
+defaultRouter.route("/statistics").get(
+  asyncHandler(async (req, res) => {
+    const totalUsers = await ApplicationModel.countDocuments({});
+    const submittedUsers = await ApplicationModel.countDocuments({
+      applicationSubmitTime: { $exists: true, $ne: null },
+    });
+    const confirmedUsers = await ApplicationModel.countDocuments({
+      confirmationBranch: { $exists: true, $ne: null },
+    });
+    let applications = {};
+    await (
+      await BranchModel.find({ type: BranchType.APPLICATION })
+    ).forEach(async branch => {
+      const branchName = branch.name;
+      const count = await ApplicationModel.countDocuments({
+        applicationBranch: branch,
+      });
+      applications = { ...applications, [branchName]: count };
+    });
+
+    let confirmations = {};
+    await (
+      await BranchModel.find({ type: BranchType.CONFIRMATION })
+    ).forEach(async branch => {
+      const branchName = branch.name;
+      const count = await ApplicationModel.countDocuments({
+        confirmationBranch: branch,
+      });
+      confirmations = { ...confirmations, [branchName]: count };
+    });
+
+    const statistics = {
```

Comment:
A couple of things to note here. If you look at the amount of database queries you're making, it's gonna be wayy too many for our use case. In this case, you should do your best to try and compress the amount of database queries you make. Whenever you're doing backend development, the efficiency of your database queries is the most important. I would suggest looking into the mongoose documentation more or look at stack overflow to find ways to simplify and compress these queries together.

Additionally, you should never be having an async/await in a loop. You can look [here](https://eslint.org/docs/rules/no-await-in-loop) at the linting documentation for why this is bad and ways to eliminate this using `promise.all`.